static/app/views/settings/organizationDeveloperSettings/sentryFunctionDetails.tsx

```diff
@@ -57,30 +75,47 @@ function SentryFunctionDetails(props: Props) {
 
   const handleSubmitSuccess = data => {
     addSuccessMessage(t('Sentry Function successfully saved.', data.name));
+    const baseUrl = `/settings/${orgId}/developer-settings/sentry-functions/`;
+    // TODO: should figure out where to redirect this
+    const url = `${baseUrl}${data.slug}/`;
+    if (sentryFunction) {
+      addSuccessMessage(t('%s successfully saved.', data.name));
+    } else {
+      addSuccessMessage(t('%s successfully created.', data.name));
+    }
+    browserHistory.push(url);
   };
 
   function handleEditorChange(value, _event) {
-    form.current.setValue('code', value);
+    setCode(value);
+  }
+
+  async function handleDelete() {
+    // figure out how to send a delete request
+    await api.requestPromise(endpoint, {
+      method: 'DELETE',
+    });
+    browserHistory.push(`/settings/${orgId}/developer-settings/`);
   }
 
-  const defaultCode = `exports.yourFunction = (req, res) => {
-    let message = req.query.message || req.body.message || 'Hello World!';
-    console.log('Query: ' + req.query);
-    console.log('Body: ' + req.body);
-    res.status(200).send(message);
-};`;
   return (
     <div>
       <Feature features={['organizations:sentry-functions']}>
         <h1>{t('Sentry Function Details')}</h1>
-        <h2>{props.params.orgId}</h2>
+        <h2>
+          {sentryFunction ? "Editing  '" + sentryFunction.name + "'" : 'New Function'}
+        </h2>
         <Form
           apiMethod={method}
           apiEndpoint={endpoint}
           model={form.current}
           onPreSubmit={() => {
+            form.current.setValue('code', code);
```

Comment:
@vpandiarajan20 Instead of doing this which only happens on `onPreSubmit`, we should add an `onChange` handler to the input for the code and just update state when that happens. Otherwise we might see weird issues with state.